ddtrace/contrib/asgi/middleware.py

```diff
@@ -128,12 +128,21 @@ async def __call__(self, scope, receive, send):
         if sample_rate is not None:
             span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, sample_rate)
 
+        host_header = None
+        for key, value in scope["headers"]:
+            if key == b"host":
+                host_header = value.decode("latin-1")
+                break
+
         method = scope.get("method")
         server = scope.get("server")
-        if server and len(server) == 2:
+        scheme = scope.get("scheme", "http")
+        full_path = scope.get("root_path", "") + scope.get("path", "")
+        if host_header is not None:
+            url = f"{scheme}://{host_header}{full_path}"
+        elif server and len(server) == 2:
             port = server[1]
             server_host = server[0] + (":" + str(port) if port is not None and port != 80 else "")
```

Comment:
I know this is how it currently is, but do we need to check for `https`/`443` here?